orchestrator-salt/src/main/resources/salt-common/salt/monitoring/settings.sls

```diff
@@ -66,9 +66,29 @@
   {% set cm_cluster_type = "COMPUTE_CLUSTER" %}
 {% endif %}
 
+{% set platform = salt['pillar.get']('telemetry:platform') %}
+{% set instance_name = salt['grains.get']('fqdn') %}
+{% if platform == "AWS" %}
+     {% set awsInstanceId = salt.cmd.run('curl -s -H Metadata:true http://169.254.169.254/latest/meta-data/instance-id') %}
```

Comment:
What if the cURL is failing in the inner call? Will the entire state failed? From the condition set it looks like then it should fallback to the value from the grains, but instead of that the entire state will fail.
I think this logic needs to be separated in a script where the error handling needs to be implemented.

cc: @sodre90

Reply:
yes it fails, make sense to put that into a script...note that for long term (for paas monitoring) probably it would be ideal if the state would fail as providing hostname for our monitoring system from customer env could be problematic. but for now, fallback could be fine